File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    var answer = 0;
    const N = numbers.length;
    
    function dfs(sum,idx){
        if(idx===N){
            if(sum===target)    answer++;
            return;
        }
        
        dfs(sum+numbers[idx], idx+1);
        dfs(sum-numbers[idx], idx+1);
    }
    
    dfs(0, 0);
    
    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    N = len(numbers)
    
    def dfs(total, idx):
        nonlocal answer
        if idx == N:
            if total == target:
                answer += 1
            return;
        
        dfs(total+numbers[idx], idx+1)
        dfs(total-numbers[idx], idx+1)
    
    dfs(0, 0)
    
    return answer